pkg/agent/agent_windows.go

```diff
@@ -187,3 +201,56 @@ func (i *Initializer) initHostNetworkFlows() error {
 func (i *Initializer) getTunnelPortLocalIP() net.IP {
 	return i.nodeConfig.NodeIPAddr.IP
 }
+
+// getHostRoutes returns routes which are configured on uplink interface before
+// the interface the configured as the uplink of antrea HNS network.
+// The routes will be restored on OVS bridge local interface after the IP configuration
+// is moved to the OVS bridge.
+func (i *Initializer) getHostRoutes() ([]netroute.Route, error) {
+	nr := netroute.New()
+	defer nr.Exit()
+	routes, err := nr.GetNetRoutesAll()
+	if err != nil {
+		return nil, err
+	}
+	var hostRoutes []netroute.Route
+	for _, route := range routes {
+		if route.LinkIndex != i.nodeConfig.UplinkNetConfig.Index {
+			continue
+		}
+		if route.GatewayAddress.String() != i.nodeConfig.UplinkNetConfig.Gateway {
+			continue
+		}
+		// Skip IPv6 routes before we support IPv6 stack.
+		if route.DestinationSubnet.IP.To4() == nil {
+			continue
+		}
+		// Skip default route. The default route will be added automatically when
+		// configure IP address on OVS bridge local interface.
+		if route.DestinationSubnet.IP.IsUnspecified() {
+			continue
+		}
```

Comment:
In my mind, there should be also some other routes added along with a new address configured on a specified interafce, like using the multicast address, or the 32bit IP address of the uplink interface as the destination. Will you leave them in the routes slice and restore?

Reply:
These routes won't be restored. The value of gateway of mentioned routes is "0.0.0.0", so these routes will be skiped by "route.GatewayAddress.String() != i.nodeConfig.UplinkNetConfig.Gateway".